### PR TITLE
ui-if directive with tests

### PR DESCRIPTION
**Pull request by ajoslin**
Taken from https://github.com/tigbro/jquery-mobile-angular-adapter/blob/master/src/main/webapp/utils/if.js

### CODE REVIEW

File: modules/directives/if/if.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+angular.module('ui.directives').directive('uiIf', [function() {
+  return {
+    transclude: 'element',
+    priority: 1000,
+    terminal: true,
+    restrict: 'A',
+    compile: function(element, attr, linker) {
+      return function(scope, iterStartElement, attr) {
+        iterStartElement[0].doNotMove = true;
```
Comment (boneskull) [inline modules/directives/if/if.js:14]: What is this?

Comment (ajoslin) [inline modules/directives/if/if.js:14]: I have no idea.... 
I can't seem to find any docs on it.

**Comment by ajoslin** [comment]
Committed with another repeater, it still works.  I think it _does_ add a comment.

**Comment by ajoslin** [comment]
Is this good to merge?  Should I make a page on angular-ui.github.com for it?

**Pull request merged**